src/AppContext.jsx

```diff
@@ -0,0 +1,295 @@
+import React, { useState, useEffect, createContext } from 'react';
+import Stopwatch from './components/timers/Stopwatch';
+import Countdown from './components/timers/Countdown';
+import XY from './components/timers/XY';
+import Tabata from './components/timers/Tabata';
+
+export const TimerContext = createContext({});
+
+export const AppContext = ({ children }) => {
+  const [timers, setTimers] = useState([]);
+  const [timersChanged, setTimersChanged] = useState(0);
+  const [appTimerAction, setAppTimerAction] = useState('');
+  const [appTimerIndex, setAppTimerIndex] = useState(-1);
+
+  // Stopwatch states
+  const [stopwatchTime, setStopwatchTime] = useState(0);
+  const [stopwatchTargetValue, setStopwatchTargetValue] = useState(0);
+  const [stopwatchIsRunning, setStopwatchIsRunning] = useState(false);
+  const [stopwatchIsPaused, setStopwatchIsPaused] = useState(false);
+
+  // Countdown states
+  const [countdownTime, setCountdownTime] = useState(0);
+  const [countdownValue, setCountdownValue] = useState(0);
+  const [countdownIsRunning, setCountdownIsRunning] = useState(false);
+  const [countdownIsPaused, setCountdownIsPaused] = useState(false);
+
+  // XY states
+  const [xyTime, setXyTime] = useState(0);
+  const [xyTimeValue, setXyTimeValue] = useState(0);
+  const [xyRounds, setXyRounds] = useState(2);
+  const [xyCurrentRound, setXyCurrentRound] = useState(1);
+  const [xyIsRunning, setXyIsRunning] = useState(false);
+  const [xyIsPaused, setXyIsPaused] = useState(false);
+
+  // Tabata states
+  const [tabataCountdown, setTabataCountdown] = useState(0);
+  const [tabataRestdown, setTabataRestdown] = useState(0);
+  const [tabataTotalCountdown, setTabataTotalCountdown] = useState(0);
+  const [tabataTotalRestdown, setTabataTotalRestdown] = useState(0);
+  const [tabataRounds, setTabataRounds] = useState(0);
+  const [tabataCurrentRound, setTabataCurrentRound] = useState(1);
+  const [tabataIsRunning, setTabataIsRunning] = useState(false);
+  const [tabataIsPaused, setTabataIsPaused] = useState(false);
+
+  useEffect(() => {}, [timersChanged]);
+
+  const getNextValidIndex = (index) => {
+    for (let i = index + 1; i < timers.length; i++) {
+      if (timers[i].valid) {
+        return i;
+      }
+    }
+    return false;
+  };
+
+  const appControl = (value) => {
+    if (value === 'Reset') {
```

Comment:
Your reset should reset the timer to its initial values, preferably not 0. If a user wants to re-run a workout, they have to re-enter their inputs.